lib/fs_utils/asset.js

```diff
@@ -45,7 +45,7 @@ class Asset {
   copy() {
     const path = this.path;
     if (isIgnored(path)) return Promise.resolve();
-    return copyFile(path, this.destPath).then(() => {
+    return copyFile(path, this.destinationPath).then(() => {
```

Comment:
I've actually already fixed this one in https://github.com/brunch/brunch/commit/3f20ff4d3571644794717572c8b75c52a37f770f

Combining a fix and adding some test structure in a single PR is probably not the best thing... at least to get the fix in asap :) That said, I'll look at the tests portion now